libs/azure-iot-middleware-freertos/libraries/azure-sdk-for-c/sdk/src/azure/iot/az_iot_adu_client.c

```diff
@@ -240,12 +237,21 @@ AZ_NODISCARD az_result az_iot_adu_client_get_agent_state_payload(
 
   _az_RETURN_IF_FAILED(az_json_writer_append_end_object(ref_json_writer));
 
-  /* Fill the compatible property names. */
+  /* Fill the compatibility property names. */
   _az_RETURN_IF_FAILED(az_json_writer_append_property_name(
       ref_json_writer,
       AZ_SPAN_FROM_STR(AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_COMPAT_PROPERTY_NAMES)));
-  _az_RETURN_IF_FAILED(az_json_writer_append_string(
-      ref_json_writer, AZ_SPAN_FROM_STR(AZ_IOT_ADU_CLIENT_AGENT_COMPATIBILITY)));
+  if (az_span_is_content_equal(
+          device_properties->compatibility_properties, AZ_SPAN_EMPTY))
+  {
+    _az_RETURN_IF_FAILED(az_json_writer_append_string(
+        ref_json_writer, AZ_SPAN_FROM_STR(AZ_IOT_ADU_CLIENT_AGENT_DEFAULT_COMPATIBILITY_PROPERTIES)));
+  }
```

Comment:
for this here, should we allow it to be empty? The user should either have their own or use the default. But in either case we want them to make the selection. Kind of like the sample from above in the middleware. We show that people should use the default "by default".

Reply:
Done.